src/main/scala/mesosphere/marathon/core/task/Task.scala

```diff
@@ -43,16 +70,37 @@ case class Task(
 }
 
 object Task {
+  def tasksById(tasks: Iterable[Task]): Map[Task.Id, Task] = tasks.iterator.map(task => task.taskId -> task).toMap
+
   case class Id(idString: String) {
+    // FIXME(PK): For dynamic reservations, the mesosTaskId has to take the launchCounter into
```

Comment:
Confusing...
Task.Id can be either the taskId (simple task) or the static part of a reserved task?